tests/test_restrictions.py

```diff
@@ -34,126 +36,130 @@ def init_class_fixtures(cls):
-                    (str_to_ts('2011-01-04' + hour_of_day), FROZEN),
-                ],
-                self.ASSET2: [
-                    (str_to_ts('2011-01-06' + hour_of_day), ALLOWED),
-                    (str_to_ts('2011-01-05' + hour_of_day), FROZEN),
-                    (str_to_ts('2011-01-07' + hour_of_day), FROZEN),
-                ],
-            }
-
-        restrictions = sum([
-            [Restriction(asset, info[0], info[1]) for info in r_history]
-            for asset, r_history in iteritems(restriction_dates)
-        ], [])
-        rl = HistoricalRestrictions(restrictions)
+            maybe_scramble = lambda r: r
+
+        def rdate(s):
+            """Convert a date string into a restriction for that date."""
+            # Add date_offset to check that we handle intraday changes.
+            return str_to_ts(s) + date_offset
+
+        all_restrictions = (
+            maybe_scramble([
+                Restriction(self.ASSET1, rdate('2011-01-04'), FROZEN),
+                Restriction(self.ASSET1, rdate('2011-01-05'), ALLOWED),
+                Restriction(self.ASSET1, rdate('2011-01-06'), FROZEN),
+            ])
+            +
+            maybe_scramble([
+                Restriction(self.ASSET2, rdate('2011-01-05'), FROZEN),
+                Restriction(self.ASSET2, rdate('2011-01-06'), ALLOWED),
+                Restriction(self.ASSET2, rdate('2011-01-07'), FROZEN),
+            ])
+        )
+        restrictions_by_asset = groupby(lambda r: r.asset, all_restrictions)
 
+        rl = HistoricalRestrictions(all_restrictions)
         assert_not_restricted = partial(self.assert_not_restricted, rl)
         assert_is_restricted = partial(self.assert_is_restricted, rl)
-        assert_vectorized_results = partial(self.assert_vectorized_results, rl)
+        assert_all_restrictions = partial(self.assert_all_restrictions, rl)
 
-        for asset, r_history in iteritems(restriction_dates):
-            dts = sorted([info[0] for info in r_history])
+        # Check individual restrictions.
+        for asset, r_history in iteritems(restrictions_by_asset):
+            freeze_dt, unfreeze_dt, re_freeze_dt = (
+                sorted([r.effective_date for r in r_history])
+            )
 
             # Not restricted until on or after the freeze
-            assert_not_restricted(asset, dts[0] - MINUTE)
-            assert_is_restricted(asset, dts[0])
-            assert_is_restricted(asset, dts[0] + MINUTE)
+            assert_not_restricted(asset, freeze_dt - MINUTE)
+            assert_is_restricted(asset, freeze_dt)
+            assert_is_restricted(asset, freeze_dt + MINUTE)
 
-            # Unrestricted on or after the unfreeze
-            assert_is_restricted(asset, dts[1] - MINUTE)
-            assert_not_restricted(asset, dts[1])
-            assert_not_restricted(asset, dts[1] + MINUTE)
+            # Unrestricted until on or after the unfreeze
```

Comment:
I don't think 'until' here is right. It was restricted before the unfreeze, and unrestricted on or after the unfreeze. Until the unfreeze, it was still restricted